shared/Microsoft.Extensions.TypeNameHelper.Sources/TypeNameHelper.cs

```diff
@@ -104,29 +192,36 @@ private static void ProcessNestedGenericTypes(Type t, StringBuilder sb, bool ful
                 {
                     var name = part.Substring(0, num);
                     var numberOfGenericTypeArgs = int.Parse(part.Substring(num + 1));
-                    if (fullName || i == totalParts - 1)
+                    if (isFullName || i == totalParts - 1)
                     {
                         sb.Append(name);
-                        AppendGenericArguments(genericArguments, index, numberOfGenericTypeArgs, sb, fullName);
+                        AppendGenericArguments(genericArguments, index, numberOfGenericTypeArgs, sb, nameFormatting);
                     }
-                    if (fullName && i != totalParts - 1)
+                    if (isFullName && i != totalParts - 1)
                     {
                         sb.Append("+");
                     }
                     index += numberOfGenericTypeArgs;
                 }
                 else
                 {
-                    if (fullName || i == totalParts - 1)
+                    if (isFullName || i == totalParts - 1)
                     {
                         sb.Append(part);
                     }
-                    if (fullName && i != totalParts - 1)
+                    if (isFullName && i != totalParts - 1)
                     {
                         sb.Append("+");
                     }
                 }
             }
         }
+
+        private enum NameFormatting
```

Comment:
No Enums. What does `FullIfAvailable` represent? What is the case where FullName is no available? At least that does not happen for types.

Reply:
Situation where `FullName` is `null` possible for types that marks open generic arguments. For example `typeof(List<>).GetGenericArguments()` will return it. For such types `FullName` is null but `Name` contains the name of an argument. 

It also could be useful if we will need to output type for `typeof(Dictionary<,>)` not like:
`Dictionary`
but like:
`Dictionary<TKey,TValue>` 

`NameFormatting` enum used strictly inside of this class to prevent multiple boolean flags.

Reply:
It is semantics and something which is not user-controllable.
If the type is open generic, like `List<>` then we print `List<T>` (based on whatever decision we make on that as implementation). But there is no distinction in fullName or name. Essentially it is not user input so code should not pass it around as argument.

Reply:
Current output for open generics:
`List<>` - `List`
`Dictionary<,>` - `Dictionary`
And there is no way to output `List<T>`, its part of changes required for method display.

If we add the ability to chose generics output method in the public method should we also make enum public?

Reply:
We add functionality in TypeNameHelper which would print `List<T>` for `List<>` but we don't need any enum for that. If it is closed generic we have actual types. If it is closed generic then we put generic args.

Reply:
Oh, that's great if we can do it default behavior.

Reply:
If there is any other behavior then let us know. I couldn't not think of anything else.